src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+ *
+ * Most of the heavy lifting is done by the 'moments' js library,
+ * with the moment-with-locales and moment-timezones add-ons.
+ *
+ * By sending preferences data, even in an incomplete state, we can generate the appropriate
+ * datetime string for surfacing to the user.
+ *
+ * @module DateUtils
+ */
+
+
+/*
+DATE FORMATS:
+These were previously called from edx-platform/common/djangoapps/util/date_utils.py,
+but in an effort to move this to the front-end, they are redefined here and translated into
+local-aware moment.js formats.
+
+DEFAULT_SHORT_DATE_FORMAT
+example: Jan 01, 2016
+python: "%b %d, %Y"
+
+DEFAULT_LONG_DATE_FORMAT
+example: Friday, January 01, 2016
+python: "%A, %B %d, %Y"
+
+DEFAULT_TIME_FORMAT
+example: Jan 01, 2016 at 15:30
+python: "%b %d, %Y at %H:%M"
+
+DEFAULT_DATE_TIME_FORMAT
+example: Jan 01, 2016 15:30 UTC
+python: "%b %d, %Y %H:%M"
+*/
+
+var DATE_FORMATS = {
+    DEFAULT_SHORT_DATE_FORMAT: 'll',
+    DEFAULT_LONG_DATE_FORMAT: 'LLLL',
+    DEFAULT_TIME_FORMAT: 'LTS z',
+    DEFAULT_DATE_TIME_FORMAT: 'll HH[:]mm z'
+};
+
+
+(function(define) {
+    'use strict';
+    define([
+        'jquery',
+        'edx-ui-toolkit/js/utils/string-utils',
+        'moment',
+        'momenttimezone'
+    ], function(
+        $,
+        StringUtils,
+        moment,
+        momentTZ
+        ) {
+        var DateUtils, isValid, dateObject, userTimezone, localizeTime, timeDisplay, localize;
+
+        /**
+         * Helper function to determine variable/context validity.
+         *
+         * @param {string} candidateVariable to test
+         * @returns {boolean}
+         */
+        isValid = function(candidateVariable) {
+            var valid = true;
+            if (candidateVariable === undefined) {
+                valid = false;
+            } else if (candidateVariable === 'undefined') {
+                valid = false;
+            } else if (candidateVariable === '') {
+                valid = false;
+            } else if (candidateVariable === 'Invalid date') {
+                valid = false;
+            } else {
+                valid = true;
+            }
+            return valid;
+        };
+
+        /**
+         * Attempt to convert a string to mutable 'moment' datetime object.
+         *
+         * The input datetimeString can be just a date, a date and time,
+         * or a complex date/time string with TZ offset. This is fairly wide-open, as
+         * many strings will satisfy this requirement.
+         * (e.g. 01-01-16 and Jan 1, 2016, and 2016-01-01 01:00:00+00:00 will work)
+         *
+         * The recommendation is to use a fully informed string, e.g. '2016-01-01 01:00:00+00:00',
+         * to limit possible date, time, and timezone ambiguities.
+         *
+         * @param {string} datetimeString. A string representation of a datetime.
+         * @returns {object} A moment.js UTC datetime object.
+         *
+        */
+        dateObject = function(datetimeString) {
```

Comment:
Right, @nasthagiri  that makes sense. I actually went with 'stringtoObject', as it's a moment-defined object, and not a proper js datetime, per se.